client/src/pages/JoinRoomPage/JoinRoomContent.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from "react";
+import JoinRoomInputs from "./JoinRoomInputs";
+import { connect } from "react-redux";
+import OnlyWithAudioCheckbox from "./OnlyWithAudioCheckbox";
+import {
+  setConnectOnlyWithAudio,
+  setIdentity,
+  setRoomId,
+} from "../../store/actions";
+import ErrorMessage from "./ErrorMessage";
+import JoinRoomButtons from "./JoinRoomButtons";
+import { useNavigate } from "react-router-dom";
+import { getRoomExists } from "../../utils/api";
+
+const JoinRoomContent = (props) => {
+  const {
+    isRoomHost,
+    setConnectOnlyWithAudio,
+    connectOnlyWithAudio,
+    setIdentityAction,
+    setRoomIdAction,
+  } = props;
+
+  const [roomIdValue, setRoomIdValue] = useState("");
+  const [nameValue, setNameValue] = useState("");
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const navigate = useNavigate();
+
+  const handleJoinRoom = async () => {
+    setIdentityAction(nameValue);
+    if (isRoomHost) {
+      createRoom();
+    } else {
+      await joinRoom();
+    }
+  };
+
+  const joinRoom = async () => {
+    const responseMessage = await getRoomExists(roomIdValue);
+
+    const { roomExists, full } = responseMessage;
+
+    if (roomExists) {
+      if (full) {
+        setErrorMessage("Meeting is full. Please try again later.");
+      } else {
+        // join a room !
+        setRoomIdAction(roomIdValue);
+        navigate("/room");
+      }
+    } else {
+      setErrorMessage("Meeting not found. Check your meeting id.");
+    }
+  };
+
+  const createRoom = () => {
+    navigate("/room");
+  };
+
+  return (
+    <>
+      <JoinRoomInputs
+        roomIdValue={roomIdValue}
+        setRoomIdValue={setRoomIdValue}
+        nameValue={nameValue}
+        setNameValue={setNameValue}
+        isRoomHost={isRoomHost}
+      />
+      <OnlyWithAudioCheckbox
+        setConnectOnlyWithAudio={setConnectOnlyWithAudio}
+        connectOnlyWithAudio={connectOnlyWithAudio}
+      />
+      <ErrorMessage errorMessage={errorMessage} />
+      <JoinRoomButtons
+        handleJoinRoom={handleJoinRoom}
+        isRoomHost={isRoomHost}
+      />
+    </>
+  );
+};
+
+const mapStoreStateToProps = (state) => {
+  return {
+    ...state,
+  };
+};
+
+const mapActionsToProps = (dispatch) => {
+  return {
+    setConnectOnlyWithAudio: (onlyWithAudio) =>
+      dispatch(setConnectOnlyWithAudio(onlyWithAudio)),
+    setIdentityAction: (identity) => dispatch(setIdentity(identity)),
+    setRoomIdAction: (roomId) => dispatch(setRoomId(roomId)),
+  };
+};

```

Comment:
Instead of using the mapStoreStateToProps and mapActionsToProps functions, you can use the mapStateToProps and mapDispatchToProps syntax directly in the connect function. This will make the code more readable. For example:

Reply:
I will use the newest version of redux to avoid that useless code in a future ✅